cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,232 @@
+
+    ```bash
+    rad resource-provider show Mycompany.Messaging
+    RESOURCE-PROVIDER   TYPE                                STATE
+    Mycompany.Messaging System.Resources/resourceProviders  Succeeded
+    ```
+
+#### 4. rad resource-provider delete
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Users can delete a resource provider.
+
+    ```bash
+    rad resource-provider delete Mycompany.Messaging
+    ```
+**Behaviors**
+Option 1 : Radius can prompt the user with a warning message if there are resources provisioned and running under the resource provider. If the user confirms deletion, Radius cleans up everything, deletes resources, types and resource providers.
+
+Option 2 : Radius can error out if there are resources provisioned and running under the resource provider. The user needs to delete the resources before deleting the resource provider.
+
+### rad resource-type
+
+#### 1. rad resource-type create
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Each resource type can define multiple API versions, and each API version defines a schema that resource instances conform to. Inputs can be passed in as a file or as flags.
+
+Create a resource type from a file 
+    
+```bash
+rad resource-type create Mycompany.Messaging/plaidResource -f plaid.yaml
+```
+
+**Flags**
+- `-f`, `--fom-file` : The file that contains the resource type definition
+
+#### 2. rad resource-type list
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Users can list all the resource types available in the resource provider.
+
+```bash
+rad resource-type list --resource-provider Mycompany.Messaging
+RESOURCE-TYPE   TYPE   STATE
+plaidResource   Mycompany.Messaging Succeeded
+resourceType2   Mycompany.Messaging Succeeded
+```
+
+#### 3. rad resource-type show
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Users can display the details of the resource types.
+
+```bash
+rad resource-type show Mycompany.Messaging/plaidResource
+RESOURCE-TYPE   TYPE   STATE
+plaidResource   Mycompany.Messaging Succeeded
+```
+
+#### 4. rad resource-type delete
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Users can delete a resource type.
+
+```bash
+rad resource-type delete Mycompany.Messaging/plaidResource
+```
+
+**Behaviors**:
+
+Option 1 : Radius can prompt the user with a warning message if there are resources provisioned and running under the resource type. If the user confirms deletion, Radius cleans up everything, deletes resources, types and resource providers.
+
+Option 2 : Radius can error out if there are resources provisioned and running under the resource type. The user needs to delete the resources before deleting the resource type.
```

Comment:
Note that there's inherently a data-race at work here if we implement this behavior in the CLI.

If other users are creating resources at the same thing, then this command might fail. I think that's OK, it's an edge case and we prevented you from data loss.